src/run_program.rs

```diff
@@ -329,15 +329,27 @@ impl<'a, D: Dialect> RunProgramContext<'a, D> {
 }
 
 pub fn run_program<'a, D: Dialect>(
+    allocator: &'a mut Allocator,
```

Comment:
maybe `run_program_with_program_arg_pair` ? The thing being passed in is not what I think of as a program, but rather a program/argument pair.